src/ringmesh/geomodel/builder/geomodel_builder.cpp

```diff
@@ -666,17 +688,39 @@ namespace
             }
         }
 
-        bool propagate( index_t t ) const
+        bool propagate( index_t t, bool backward ) const
         {
             return t != cur_border_polygon_ && !is_visited( t )
-                   && equal_to_line_adjacent_surfaces( t );
+                && equal_to_line_adjacent_surfaces( t ) && !vertex_is_on_corner( t, backward );
         }
 
         bool is_visited( index_t i ) const
         {
             return visited_[i];
         }
 
+        bool vertex_is_on_corner( index_t t, bool backward ) const
+        {
+            const auto& surfaces = [t, backward, this]{
+                if( backward )
+                {
+                    return surfaces_around_vertices_[this->border_polygons_[t].v1];
```

Comment:
you are returning as bool a vector of index_t ?

Reply:
No, I am returning a vector of index_t inside the variable surfaces